sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample10_MockingReadingEventsTests.cs

```diff
@@ -0,0 +1,290 @@
+                .Setup(c => c.GetPartitionIdsAsync(It.IsAny<CancellationToken>()))
+                .ReturnsAsync(new string[] { mockPartitionId });
+            mockEventHubConsumerClient
+                .Setup(c => c.ReadEventsFromPartitionAsync(It.IsAny<string>(), It.IsAny<EventPosition>(), It.IsAny<CancellationToken>()))
+                .Returns(GetMockEventsFromPartition(mockPartitionId));
+
+            var consumer = mockEventHubConsumerClient.Object;
+            try
+            {
+                using CancellationTokenSource cancellationSource = new CancellationTokenSource();
+                cancellationSource.CancelAfter(TimeSpan.FromSeconds(30));
+
+                string firstPartition = (await consumer.GetPartitionIdsAsync(cancellationSource.Token)).First();
+                EventPosition startingPosition = EventPosition.Earliest;
+
+                await foreach (PartitionEvent partitionEvent in consumer.ReadEventsFromPartitionAsync(
+                    firstPartition,
+                    startingPosition,
+                    cancellationSource.Token))
+                {
+                    string readFromPartition = partitionEvent.Partition.PartitionId;
+                    ReadOnlyMemory<byte> eventBodyBytes = partitionEvent.Data.EventBody.ToMemory();
+
+                    Debug.WriteLine($"Read event of length { eventBodyBytes.Length } from { readFromPartition }");
+                }
+            }
+            catch (TaskCanceledException)
+            {
+                // This is expected if the cancellation token is
+                // signaled.
+            }
+            finally
+            {
+                await consumer.CloseAsync();
+            }
+
+            #endregion
+        }
+
+        /// <summary>
+        ///   Performs basic unit test validation of the contained snippet.
+        /// </summary>
+        ///
+        [Test]
+        public async Task ReadPartitionWithReceiver()
+        {
+            #region Snippet:EventHubs_Sample10_MockingReadPartitionWithReceiver
+
+            var mockEventHubProducerClient = new Mock<EventHubProducerClient>();
+            mockEventHubProducerClient
+                .Setup(c => c.GetPartitionIdsAsync(It.IsAny<CancellationToken>()))
+                .ReturnsAsync(new string[1] { "sample id" });
+
+            var mockEventHubPartitionReceiver = new Mock<PartitionReceiver>();
+            var mockBatchData = GenerateBatchData();
+            mockEventHubPartitionReceiver
+                .Setup(r => r.ReceiveBatchAsync(It.IsAny<int>(), It.IsAny<TimeSpan>(), It.IsAny<CancellationToken>()))
+                .ReturnsAsync(mockBatchData)
+                .Verifiable();
+
+            using CancellationTokenSource cancellationSource = new CancellationTokenSource();
+            cancellationSource.CancelAfter(TimeSpan.FromSeconds(30));
+
+            string firstPartition;
+
+            await using (var producer = mockEventHubProducerClient.Object)
```

Comment:
I think we can bypass this part, since we're mocking the receiver directly.   Since we control the partition that we're accepting, then we don't need to show the flow of querying one.